app/helpers/application_helper.rb

```diff
@@ -260,6 +244,18 @@ def view_to_url(view, parent = nil)
     end
   end
 
+  def class_to_db(cname, parent)
+    return unless cname
+    case
```

Comment:
afaik this can be shortened by doing `case cname` and removing repeating `cname` after each 'when' keyword